### PR TITLE
add flag kernel-memory (NOP)

### PR DESCRIPTION
**Pull request by ningmingxiao**
Signed-off-by: ningmingxiao <ning.mingxiao@zte.com.cn>

### CODE REVIEW

File: cmd/nerdctl/run_cgroup_linux_test.go [hunk 1]
```diff
@@ -152,13 +159,14 @@ func TestRunCgroupV1(t *testing.T) {
...
+	const expected = "42000\n100000\n0\n44040192\n6291456\n104857600\n0\n52428800\n42\n2000\n0-1\n"
+	base.Cmd("run", "--rm", "--cpus", "0.42", "--cpuset-mems", "0", "--memory", "42m", "--memory-reservation", "6m", "--memory-swap", "100m", "--memory-swappiness", "0", "--kernel-memory", "50m", "--pids-limit", "42", "--cpu-shares", "2000", "--cpuset-cpus", "0-1", testutil.AlpineImage, "cat", quota, period, cpusetMems, memoryLimit, memoryReservation, memorySwap, memorySwappiness, kernelMemory, pidsLimit, cpuShare, cpusetCpus).AssertOutExactly(expected)
+	base.Cmd("run", "--rm", "--cpu-quota", "42000", "--cpu-period", "100000", "--cpuset-mems", "0", "--memory", "42m", "--memory-reservation", "6m", "--memory-swap", "100m", "--memory-swappiness", "0", "--kernel-memory", "50m", "--pids-limit", "42", "--cpu-shares", "2000", "--cpuset-cpus", "0-1", testutil.AlpineImage, "cat", quota, period, cpusetMems, memoryLimit, memoryReservation, memorySwap, memorySwappiness, kernelMemory, pidsLimit, cpuShare, cpusetCpus).AssertOutExactly(expected)
```
[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply cmd/nerdctl/run_cgroup_linux_test.go:169]: > It is cgroup v1 testcase, cgroupv1 is support in runc

No, the kernel-memory support has been completely removed from runc.
https://github.com/opencontainers/runc/blob/eddf35e5462e2a9f24d8279874a84cfc8b8453c2/libcontainer/cgroups/fs/memory.go#L116

[the remark this answers is not in the document]
Comment (AkihiroSuda) [reply cmd/nerdctl/run_cgroup_linux_test.go:169]: The flag can be implemented as a NOP flag and should print a warning

[the remark this answers is not in the document]
Comment (ningmingxiao) [reply cmd/nerdctl/run_cgroup_linux_test.go]: done

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Pull request merged**